fdbserver/DataDistribution.actor.cpp

```diff
@@ -1686,14 +1916,25 @@ ACTOR Future<Void> doAuditOnStorageServer(Reference<DataDistributor> self,
 		    .detail("Range", req.range)
 		    .detail("StorageServer", ssi.toString())
 		    .detail("TargetServers", describe(req.targetServers));
-		if (e.code() == error_code_actor_cancelled || e.code() == error_code_audit_storage_error) {
+		if (e.code() == error_code_actor_cancelled) {
 			throw e;
+		} else if (e.code() == error_code_audit_storage_error) {
+			audit->foundError = true;
+		} else if (audit->retryCount > SERVER_KNOBS->AUDIT_RETRY_COUNT_MAX) {
+			throw audit_storage_failed();
+		} else if (e.code() == error_code_transaction_too_old) {
```

Comment:
It could be recoverable, i.e., some SS falls behind but catches up soon, what's the reason we set it as failed early?